server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/manager.js

```diff
@@ -25,29 +25,52 @@
 
 var managerStatusTable, recoveryListTable;
 
-function refreshManagerBanner() {
+function refreshManagerBanners() {
   getStatus().then(function () {
-    var managerStatus = JSON.parse(sessionStorage.status).managerStatus;
+    const managerStatus = JSON.parse(sessionStorage.status).managerStatus;
 
     // If manager status is error
     if (managerStatus === 'ERROR') {
-      // show banner and hide table
-      $('#managerBanner').show();
+      // show the manager error banner and hide table
+      $('#managerRunningBanner').show();
       $('#managerStatus_wrapper').hide();
     } else {
-      // otherwise, hide banner and show table
-      $('#managerBanner').hide();
+      // otherwise, hide the error banner and show manager table
+      $('#managerRunningBanner').hide();
       $('#managerStatus_wrapper').show();
     }
   });
+
+  getManager().then(function () {
+    const managerData = JSON.parse(sessionStorage.manager);
+    const managerState = managerData.managerState;
+    const managerGoalState = managerData.managerGoalState;
+
+    const isStateGoalDifferent = managerState !== managerGoalState;
+
+    // if the manager state is normal and the goal is the same as the state or manager is not running, return early
+    if ((managerState === 'NORMAL' && !isStateGoalDifferent) || managerState === null) {
```

Comment:
Is `managerState` always `null` when the `managerStatus` is `ERROR`, like in the previous block? It seems like that's not always the case.

Reply:
Yea I think so. If managerStatus (which is pulled from `/rest/status`) is ERROR, the manager is down which makes most of the stats at `/rest/manager` become `null`.

Reply:
The issue is that they come from two different endpoints, and are therefore fetched separately. There is no reason to think they are always in sync.

Reply:
The new code does not rely on any data from `/rest/status`. The two blocks in `refreshManagerBanners()` each pull from different endpoints and act independently of each other.

Reply:
> The new code does not rely on any data from `/rest/status`. The two blocks in `refreshManagerBanners()` each pull from different endpoints and act independently of each other.

Right, that's my point. There are two banners. The other banner pulls from the other endpoint. The end result is that it could say something in one banner that contradicts the other banner. For example, it could say that it it's in safe mode *and* that it's not running at the same time.

Reply:
I think we don't need to worry about this edge case too much unless we start seeing it a lot.